library/zitilib.c

```diff
@@ -1239,27 +1266,36 @@ int Ziti_resolve(const char *host, const char *port, const struct addrinfo *hint
                 res->ai_protocol = 0;
                 break;
             default: // no other protocols are supported
-                return -1;
+                goto error;
+        }
+
+        switch (hints->ai_family) {
+            case AF_INET:
+            case AF_INET6:
+            case AF_UNSPEC:
+                res->ai_family = hints->ai_family;
+                break;
+            default: // unsupported address family
+                goto error;
         }
     }
 
-    struct sockaddr_in *addr4 = calloc(1, sizeof(struct sockaddr_in6));
-    int rc = 0;
-    if ((rc = uv_ip4_addr(host, portnum, addr4)) == 0) {
-        ZITI_LOG(DEBUG, "host[%s] port[%s] rc = %d", host, port, rc);
+    if ((res->ai_family == AF_UNSPEC || res->ai_family == AF_INET)
+        && uv_ip4_addr(host ? host : "127.0.0.1", portnum, &addr->in4) == 0) {
```

Comment:
if `host == NULL` we should just use `INADDR_LOOPBACK` or `INADDR_ANY` (see my other comment)
to avoid parsing

Reply:
Hi @ekoby -

Reflecting on the ziti security model, does it make sense to allow a socket to bind to ```INADDR_ANY``` by default? If the user is zitifying, should we stop insecure practices?

Reply:
Turned out that ```sin_len``` checks are necessary as we are directly manipulating ```sockaddr_in``` members.

Reply:
> Hi @ekoby -
> 
> Reflecting on the ziti security model, does it make sense to allow a socket to bind to `INADDR_ANY` by default? If the user is zitifying, should we stop insecure practices?

`zitify` should not break current behavior of an app if connect/bind address is not something user is trying to zitify.